### DISCUSSION TITLE
Use patheffects without importing matplotlib

### DISCUSSION THREAD

**Discussion by buhtz** [mwaskom/seaborn#3184] [Q&A]
There are known sideeffects importing `matplotlib` when using `seaborn`.
So I try to avoid `import matplotlib` and just do `import seaborn as sns`.

Now I need to add path effects to text objects in a plot.
```
# type(text) -> <class 'matplotlib.text.Text'>
text.set_path_effects(
            [
                matplotlib.patheffects.path_effects.Stroke(
                    linewidth=3,
                    foreground=median.get_color()),
                matplotlib.patheffects.path_effects.Normal(),
            ]
        )
```

You see I have to `import matplotlib.patheffects` here. Can I prevent this and find a seaborn-only solution for that?

Here is an example of how the path effects looking in real
![image](https://user-images.githubusercontent.com/11861496/207294494-476a03b2-6152-49dc-baac-f3edee280746.png)

**Comment by mwaskom** [comment]
I'm not certain what you're hoping to achieve here (what "known side effects" are you talking about?) but there's no way to import seaborn without importing matplotlib; seaborn is an interface to matplotlib.

**Reply by buhtz** [reply]
> (what "known side effects" are you talking about?)

It is about multiple empty plots/figures because the figure wasn't closed before.
e.g. https://stackoverflow.com/q/52048553/4865723
https://stackoverflow.com/q/51916372/4865723

> but there's no way to import seaborn without importing matplotlib

I meant explicit import. Of course, Seaborn do use matplotlib in the back.  Because of that the used don't need to make an extra `import matplotlib`.

**Reply by mwaskom** [reply]
Those stackoverflow questions don't have anything to do with import side effects, and in any case, import side effects wouldn't depend on whether a module was explicitly or implicitly imported.

**Reply by buhtz** [reply]
You often see in examples `plt.show()`. But you have to import `plt` from `matplotlib` first to do this.
When using Seaborn you don't need to do this. `myplot = sns.catplot(...); myplot.figure.show()` and you are fine. No need for explicit importing `matplotlib`

Let me try this question: Can I have path effects without writing something like `matplotlib.*` ?
